src/wh_server_crypto.c

```diff
@@ -1041,16 +1046,179 @@ int wh_Server_HandleCryptoRequest(whServerContext* server,
                 }
 #endif
                 break;
+#endif /* !NO_SHA256 */
 
             default:
                 ret = NOT_COMPILED_IN;
                 break;
         }
         break;
 
+    case WC_ALGO_TYPE_NONE:
+    default:
+        ret = NOT_COMPILED_IN;
+        break;
+    }
+
+    /* Propagate error code to client in response packet */
+    packet->rc = ret;
+
+    if (ret != 0)
+        *size = WH_PACKET_STUB_SIZE + sizeof(packet->rc);
+
+    /* Since crypto error codes are propagated to the client in the response
+     * packet, return success to the caller unless a cancellation has occurred
+     */
+    if (ret != WH_ERROR_CANCEL) {
+        ret = 0;
+    }
+    return ret;
+}
+
+#ifdef WOLFHSM_CFG_DMA
+
+#ifndef NO_SHA256
+static int hsmCryptoSha256Dma(whServerContext* server, whPacket* packet,
+                              uint16_t* size)
+{
+    int ret = 0;
+#if WH_DMA_IS_32BIT
+    wh_Packet_hash_sha256_Dma32_req* req = &packet->hashSha256Dma32Req;
+    wh_Packet_hash_sha256_Dma32_res* res = &packet->hashSha256Dma32Res;
+#else
+    wh_Packet_hash_sha256_Dma64_req* req = &packet->hashSha256Dma64Req;
+    wh_Packet_hash_sha256_Dma64_res* res = &packet->hashSha256Dma64Res;
+#endif
+    wc_Sha256* sha256 = server->crypto->algoCtx.sha256;
+    int        clientDevId;
+
+    /* Ensure state sizes are the same */
+    if (req->state.sz != sizeof(*sha256)) {
+        res->dmaCryptoRes.badAddr = req->state;
+        return WH_ERROR_BADARGS;
+    }
+
+    /* Copy the SHA256 context from client address space */
+    ret = whServerDma_CopyFromClient(server, sha256, req->state.addr,
```

Comment:
Saw that and it will definitely catch the problem, but there's no real indication how to fix it.  Almost want to return a stronger bug than bad args.  Like some sort of internal fault.  Not the user's fault but ours.  Let's keep this moving forward and mark this as needs improvement.  We'll tweak it as we add more DMA's in and get smarter about how to do this.